### PR TITLE
docs: add troubleshoot for failing to create emr application in the specified subnets

### PR DESCRIPTION
**Pull request by zxkane**
specified subnets

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.40|< 80|
|Duplicated lines density|:white_check_mark: OK|5.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|1|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1336)
###### _updated: 5/19/2024, 08:38:11 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1336&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1336&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1336&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

File: docs/en/troubleshooting.md [hunk 1]
```diff
@@ -151,6 +151,15 @@ It's caused by the large volume of data to be loaded or the Redshift load being
 
 The dashboard is likely using views or materialized views that are not yet ready in Redshift. The solution asynchronously creates or updates the views and materialized views used by the dashboard after creating or updating the pipeline of the project. The duration of the creation or update depends on the workload of the Redshift cluster and the existing data volume and can take minutes to hours. You can keep track of progress in the **Redshift Schemas** section in the **Processing** tab of the Pipeline Detail page. If the post-configuration job fails, you can access the execution of the workflow through its link and rerun the job via **Actions - Redrive** or **New execution** with the input unchanged.
 
+## Problem: When configuring the data pipeline with data processing enabled, the data processing CloudFormation stack encountered the error "The availability zones for subnets [subnet-xyz] are currently not supported.".
```
Comment (luorobin-a2z) [inline docs/en/troubleshooting.md:154]: --> ##Problem: When creating the data pipeline with data processing enabled, the data processing CloudFormation stack encountered error of "The availability zones for subnets [subnet-xyz] are currently not supported.".

File: docs/en/troubleshooting.md [hunk 2]
```diff
@@ -151,6 +151,15 @@ It's caused by the large volume of data to be loaded or the Redshift load being
...
+
+The EMR serverless application for processing the clickstream data cannot be deployed in the specified private subnets mentioned in the above error message.
```
Comment (luorobin-a2z) [inline docs/en/troubleshooting.md:156]: --> This is due to the EMR serverless application for processing the clickstream data cannot be deployed in the specified private subnets mentioned in the above error message.

File: docs/zh/troubleshooting.md [hunk 3]
```diff
@@ -159,6 +159,15 @@ aws logs put-resource-policy --policy-name AWSLogDeliveryWrite20150319 \
 
 仪表板可能正在使用尚未在 Redshift 中就绪的视图或物化视图。本解决方案在创建或更新项目的管道时异步创建或更新仪表板使用的视图和物化视图。创建或更新的持续时间取决于 Redshift 集群的工作负载、已有数据量，完成更新可能需要几分钟到几小时。您可以在管道详情页面的**数据处理和建模**选项卡中的 **Redshift Schemas** 部分跟踪进度。如果后期配置作业失败，您可以通过其链接访问工作流的执行，并通过**操作 - 重启**或**新执行**(输入保持不变)来重新运行作业。
 
+## 问题: 在配置启用数据处理的数据管道时，数据处理 CloudFormation 堆栈遇到了错误 "The availability zones for subnets [subnet-xyz] are currently not supported."。
+
```
Comment (luorobin-a2z) [inline docs/zh/troubleshooting.md:163]: 问题: 在创建启用数据处理的数据管道时，数据处理 CloudFormation 堆栈遇到了错误 "The availability zones for subnets [subnet-xyz] are currently not supported."。

File: docs/zh/troubleshooting.md [hunk 4]
```diff
@@ -159,6 +159,15 @@ aws logs put-resource-policy --policy-name AWSLogDeliveryWrite20150319 \
...
+处理点击流数据的 EMR 无服务器应用程序无法部署在上述错误消息中指定的私有子网中。
```
Comment (luorobin-a2z) [inline docs/zh/troubleshooting.md:164]: 这是因为处理点击流数据的 EMR 无服务器应用程序无法部署在上述错误消息中指定的私有子网中。

**Review by luorobin-a2z — CHANGES_REQUESTED** [review]
wording update

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.40|< 80|
|Duplicated lines density|:white_check_mark: OK|5.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|1|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1336)
###### _updated: 5/20/2024, 03:32:36 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.40|< 80|
|Duplicated lines density|:white_check_mark: OK|5.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|1|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1336)
###### _updated: 5/20/2024, 03:33:55 (UTC+0)_

**Review by luorobin-a2z — APPROVED** [review]

**Pull request merged**